android/logto-client/src/main/kotlin/io/logto/sdk/android/LogtoClient.kt

```diff
@@ -21,6 +25,94 @@ open class LogtoClient(
 
     fun isAuthenticated() = idToken != null
 
+    fun getAccessToken(callback: RetrieveCallback<AccessToken>) =
+        getAccessToken(null, null, callback)
+
+    fun getAccessToken(
+        resource: String?,
+        scope: List<String>?,
+        callback: RetrieveCallback<AccessToken>,
+    ) {
+        if (!isAuthenticated()) {
+            callback.onResult(LogtoException(LogtoException.Message.NOT_AUTHENTICATED), null)
+            return
+        }
+
+        resource?.let {
+            if (logtoConfig.resource?.contains(it) == false) {
+                callback.onResult(
+                    LogtoException(LogtoException.Message.RESOURCE_IS_NOT_GRANTED).apply { detail = it }, null
+                )
+                return
+            }
+        }
+
+        val finalScope = scope ?: logtoConfig.scope
+        if (!logtoConfig.scope.containsAll(finalScope)) {
+            callback.onResult(
+                LogtoException(LogtoException.Message.SCOPES_ARE_NOT_ALL_GRANTED).apply {
+                    detail = finalScope.toString()
+                },
+                null
+            )
+            return
+        }
```

Comment:
should we just honor the scope parameter? instead of fallback

Reply:
Refer to https://www.notion.so/silverhand/fetchTokenByRefreshToken-a1bc0fd7d79a4ebe9fd4c9168ae4a4d8#cff0e6c92a724ca1a1ae488c5b07eeaa
if no scope is provided, the service will use the same scopes in auth code flow.
because we append default scopes (`openid` & `offline_access`) in auth code flow, so we will have an access token cached with that scopes.
this fallback behavior makes us use the cached access token if no scope is provided.

Reply:
i see, then this refers to the point @simeng-li mentioned

Reply:
LOG-1451